db/db_impl_write.cc

```diff
@@ -271,7 +316,9 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
     should_exit_batch_group = write_thread_.CompleteParallelMemTableWriter(&w);
   }
   if (should_exit_batch_group) {
-    versions_->SetLastSequence(last_sequence);
+    if (status.ok() && !(concurrent_prepare_ && disable_memtable)) {
```

Comment:
I seems to me we won't get here in case `concurrent_prepare_ && disable_memtable` holds, because it will be redirect to `WriteImplWALOnly()`.